### PR TITLE
Create BookDto, CreateBookRequestDto, BookController, EntityNotFoundE…

### PR DESCRIPTION
**Pull request by OlegTuryk**
…xception, BookMapper, MapperConfig, update BookService

### CODE REVIEW

**Review by padu12 — APPROVED** [review]
LGTM!

File: src/main/java/com/app/onlinebookstore/OnlineBookStoreApplication.java [hunk 1]
```diff
@@ -1,34 +1,14 @@
...
-import com.app.onlinebookstore.service.BookService;
-import java.math.BigDecimal;
 import lombok.RequiredArgsConstructor;
-import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 @RequiredArgsConstructor
```
Comment (vovasalyha) [inline src/main/java/com/app/onlinebookstore/OnlineBookStoreApplication.java:8]:
```suggestion
```

File: src/main/java/com/app/onlinebookstore/repository/BookRepository.java [hunk 2]
```diff
@@ -2,9 +2,12 @@
...
 import com.app.onlinebookstore.model.Book;
 import java.util.List;
+import java.util.Optional;
 
 public interface BookRepository {
     Book save(Book book);
 
     List<Book> findAll();
+
+    public Optional<Book> getBookById(Long id);
```
Comment (vovasalyha) [inline src/main/java/com/app/onlinebookstore/repository/BookRepository.java:12]:
```suggestion
    public Optional<Book> findById(Long id);
```

**Review by vovasalyha — APPROVED** [review]
Approving, but please change method naming in repository interface

**Pull request merged**